cmake/templates/xcode-launch-c.in

```diff
@@ -7,5 +7,5 @@ if [[ "$1" = "${CMAKE_C_COMPILER}" ]] ; then
     shift
 fi
 
-export CCACHE_CPP2=true
+export CCACHE_CPP2=false
```

Comment:
That won't make things better. Try add `timeout-minutes` to https://github.com/opencv/opencv/blob/5.x/.github/workflows/PR-5.x.yaml:

```yaml
  macOS-x64:
    uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-5.x-macOS-x86_64.yaml@main
    timeout-minutes: 90
```